### ISSUE TITLE
Implement symmetric encryption with integrity

### ISSUE THREAD

**Issue by Dakkabi** [Dakkabi/Portfolio-Visualiser#14]
## Requirements
- PostgreSQL does not offer encryption on rest
- Me, as the server-hoster, should have zero knowledge / accessibility for user's data (e.g. API keys)
- Some brokers (e.g. Trading212) require the API key as a plaintext in an authorization header
- Ensure the encryption algorithms utilise an integrity checking algorithm (HMAC) to ensure there is absolutely zero data loss potential

## Deliverables
- Client-side KDF function with the user's password as input, resulting in `secret_key`, store in `sessionStorage`
- On user account creation, create a random string snippet
- Wrap using the user's `secret_key`
- Whenever data requires encryption, request user's `secret_key`
- Decrypt string digest and use result as encryption key with a symmetric encryption algorithm
- Same for decryption
- When requesting a password change, decrypt string digest using `old_secret_key`, encrypt with `new_secret_key` and store
